packages/block-library/src/buttons/block.json

```diff
@@ -8,6 +8,12 @@
 	"description": "Prompt visitors to take action with a group of button-style links.",
 	"keywords": [ "link" ],
 	"textdomain": "default",
+	"attributes": {
+		"contentContainer": {
+			"type": "null",
+			"role": "content"
+		}
+	},
```

Comment:
This isn't ideal 😳 but I couldn't think of a better way to indicate that we want Buttons to behave as a content container, to which it should always be possible to add new blocks.

Reply:
I'm not sure we should introduce yet another slightly different check in this already over-complicated logic 😅 
We already check for the content role in a few places to determine behaviour in write mode. The risk with a list of container blocks is that if only some of them have the content role, they may be treated differently in different places in the code.

It's annoying because the content role was introduced specifically [for content locking purposes](https://github.com/WordPress/gutenberg/pull/43037) so it's the ideal thing to use for this, but at the same time it's tied to being a property of block attributes so it only really works for blocks that have them. It might have been better to just make it an attribute itself.

Reply:
I think the "role": "content" still makes sense to have in content attributes that you want to be editable if the block is locked, like captions and URLs.

But we could also have a block support or even an attribute (I think I'd slightly prefer attribute because it's related to the capabilities of the block itself, rather than being a tool that can be used to modify the block, like supports usually are)

Reply:
Yeah, I was imagining it'd be both (support role:content within an attribute, and also blocks opting-in via a support).

> rather than being a tool that can be used to modify the block, like supports usually are

It looks like we have at least a few other supports that aren't design tool related, but are more about how the block can be used within the editor (i.e. `inserter`, `lock`, `multiple`, and `renaming`, and [`splitting`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#splitting) which was a TIL for me!), so I think it could sit quite nicely alongside those sorts of supports.